src/Commands/BranchNameConvention.php

```
@@ -57,37 +69,46 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
+        if (empty($_SERVER['branch_regex'])) {
+            throw new RuntimeException('Missing "branch_regex" variable in environment.');
+        }
+        if (empty($_SERVER['decline_message'])) {
+            $message = 'Invalid branch name.';
+        } else {
+            $message = $_SERVER['decline_message'];
+        }
+
         $branch = $this->config->headRef();
-        $parts = explode('/', $branch);
-        if (count($parts) !== 2) {
-            return $this->fail($output);
+        $jsonFile = $this->config->eventPath();
+        if (!file_exists($jsonFile)) {
+            throw new RuntimeException('Missing pull request event json file.');
         }
 
-        if (!in_array($parts[0], ['feature', 'bugfix', 'styling'])) {
-            return $this->fail($output);
+        $json = json_decode($jsonFile, true);
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new RuntimeException(json_last_error_msg());
         }
 
-        if (1 !== preg_match('/^[a-z]+\/[0-9]+-[a-z0-9-]+$/', $branch)) {
-            return $this->fail($output);
+        if (in_array($json['pull_request']['user']['login'], $this->codeOwnersFile->getRepositoryOwners())) {
+            // PR done by code owner, so this check is not required.
+            return 0;
+        }
+
+        if (1 !== preg_match($_SERVER['branch_regex'], $branch)) {
```

Missing the check if the variable $_SERVER['branch_regex']exists ;) Should return an error / fail.